core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/event/span/CoreEventExecutionSpanProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.core.internal.profiling.tracing.event.span;
+
+import org.mule.runtime.api.component.Component;
+import org.mule.runtime.core.api.config.MuleConfiguration;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.internal.profiling.InternalSpan;
+
+/**
+ * Provides the Span from the core event and the component.
+ *
+ * @since 4.5.0
+ */
```

Comment:
This should be a customizer of each attribute of the span (eventually increment it), not the creation of the span.